src/QueryReflection/QuerySimulation.php

```diff
@@ -129,7 +129,10 @@ public static function simulate(string $queryString): ?string
         if (null === $queryString) {
             return null;
         }
-        $queryString .= ' LIMIT 0';
+
+        if ('SELECT' === QueryReflection::getQueryType($queryString)) {
```

Comment:
```suggestion
        // make sure we don't unnecessarily transfer data, as we are only interessted in the statement is succeeding
        if ('SELECT' === QueryReflection::getQueryType($queryString)) {
```

Reply:
I think we can add the `LIMIT` to all queries except `INSERT`.
it also makes sense to limit `UPDATE`, `REPLACE` and `DELETE`.. that way we are double save to not modify real data (even though we have the transaction doing the same for us... but better save then sorry)

Reply:
done

Reply:
removed non-SLECT types again as they did not work for non-mysql